proxy_setup.sh

```diff
@@ -23,8 +23,12 @@ if [ -n "$http_proxy" ]; then
     DOMAIN="$(domainname),"
   fi
 
+  # a string like 127.0.0.1, 10.0.100.3, ..."
   local_ips=$(ip addr list |grep 'inet '|sed -e 's/.* inet //' -e 's#/.*#,#')
-  export no_proxy="$(sed 's/ //g' <<< $local_ips)$(hostname),$(hostname -f),$DOMAIN10.0.100.,10.0.100.*,$no_proxy"
+  # a string like 10.0.2.*, 10.0.100.*, ..., 10.0.2., 10.0.100., ...
+  local_nets=$(sed -e 's/127[.0-9]*, //' -e 's/\.[0-9]*,/.*,/g' <<< $local_ips)$(sed -e 's/127[.0-9]*, //' -e 's/\.[0-9]*,/.,/g' <<< $local_ips)
```

Comment:
This is a really good example of why we might wish to abandon the multiple NICs in the default setup.